core/src/main/scala/kafka/server/ClientQuotaManager.scala

```diff
@@ -374,9 +375,11 @@ class ClientQuotaManager(private val config: ClientQuotaManagerConfig,
     throttleCallback: ThrottleCallback,
     throttleTimeMs: Int
   ): Unit = {
+    // LIKAFKA-45345, even if the throttleTimeMs is 0, we still record it so that
+    // the throttle-time sensor does not expire before the byte-rate sensor
+    val clientSensors = getOrCreateQuotaSensors(request.session, request.headerForLoggingOrThrottling().clientId)
+    clientSensors.throttleTimeSensor.record(throttleTimeMs)
```

Comment:
Sorry, I've forgotten the finer details from your proposal document: will this unconditional recording of a 0 throttle time do the reverse of what it's trying to prevent, i.e., allow the byte-rate sensor to expire while the throttle-time one never does?

Reply:
After this patch, if there is traffic, both the byte-rate and throttle-time sensors perform the unconditional recording.
When traffic stops, the recording on both sensors will stop, and thus both should be able to expire.